examples/cpp/content_filter/CLIParser.hpp

```diff
@@ -0,0 +1,458 @@
+                        }
+                        else
+                        {
+                            if (config.entity == CLIParser::EntityKind::PUBLISHER)
+                            {
+                                config.pub_config.samples = static_cast<uint16_t>(input);
+                            }
+                            else if (config.entity == CLIParser::EntityKind::SUBSCRIBER)
+                            {
+                                EPROSIMA_LOG_ERROR(CLI_PARSER,
+                                        "samples option option can only be used with the Publisher");
+                                print_help(EXIT_FAILURE);
+                            }
+                            else
+                            {
+                                EPROSIMA_LOG_ERROR(CLI_PARSER, "entity not specified for --sample argument");
+                                print_help(EXIT_FAILURE);
+                            }
+                        }
+                    }
+                    catch (const std::invalid_argument& e)
+                    {
+                        EPROSIMA_LOG_ERROR(CLI_PARSER, "invalid sample argument for " + arg + ": " + e.what());
+                        print_help(EXIT_FAILURE);
+                    }
+                    catch (const std::out_of_range& e)
+                    {
+                        EPROSIMA_LOG_ERROR(CLI_PARSER, "sample argument out of range for " + arg + ": " + e.what());
+                        print_help(EXIT_FAILURE);
+                    }
+                }
+                else
+                {
+                    EPROSIMA_LOG_ERROR(CLI_PARSER, "missing argument for " + arg);
+                    print_help(EXIT_FAILURE);
+                }
+            }
+            else if (arg == "-i" || arg == "--interval")
+            {
+
+                if (i + 1 < argc)
+                {
+                    int input = std::stoi(argv[++i]);
+                    if (config.entity == CLIParser::EntityKind::PUBLISHER)
+                    {
+                        config.pub_config.interval = input;
+                    }
+                    else if (config.entity == CLIParser::EntityKind::SUBSCRIBER)
+                    {
+                        EPROSIMA_LOG_ERROR(CLI_PARSER, "interval option can only be used with the Publisher");
+                        print_help(EXIT_FAILURE);
+                    }
+                    else
+                    {
+                        EPROSIMA_LOG_ERROR(CLI_PARSER, "entity not specified for --sample argument");
+                        print_help(EXIT_FAILURE);
+                    }
+                }
+                else
+                {
+                    EPROSIMA_LOG_ERROR(CLI_PARSER, "missing argument for " + arg);
+                    print_help(EXIT_FAILURE);
+                }
+            }
+            else if (arg == "-f" || arg == "--filter")
```

Comment:
I think that we could clarify this a little bit more:
* `--filter`  is an argument for the subscriber only and defines the content filter expression.
* We can have another argument on the subscriber `--writer-side` that sets both intraprocess and multicast to OFF or ON (on the subscriber) depending if the argument is found in the cli args.
* On the publisher, we would change the `--filter` to `--reader-filters <num>`, if the argument is not defined, we use the default `reader_filters_allocation` values (which is set to the max 32 I believe)

What do you think ?

Reply:
I think they are good ideas! Only a couple of things:
* It's a great idea having an argument for the subscriber to define the content filter expression in case of default filter. But I would keep the other argument to define the kind of filter to apply (default or custom). We could have `--filter-kind` and `--filter-expression`
* I like the the idea of defining the writer side filter setting the `--reader-filters` number
* I am not really sure about the argument `--writer-side`, I think that it would not be very clear since there is not a specific way to activate the writer side filter but it is only a matter of the QoS that are used. I would rather add the reference to the documentation in the ReadMe file, where it is explained which are the conditions that allows the filter on writer side.
What do you think?

Reply:
* Good one! I like the idea
* Nice
* I think that it is still worth leaving a cli argument giving the chance of applying the required qos for writer-side-filtering (maybe we can rename it to`--writer-side-filtering`). In addition, as you were saying, we can leave a link in the README to the docs section regarding writer side filtering explaning that the flag just adds some qos that allow writer side filtering if `reader_allocation_filters` is not 0 nor it has reached the maximum number in the writer side.

Reply:
After internal discussion we decided to not add the flag, leaving only `--reader-filters` to perform or not the writer side filtering